client/transaction-pool/src/lib.rs

```diff
@@ -495,26 +498,40 @@ impl<PoolApi, Block> MaintainedTransactionPool for BasicPool<PoolApi, Block>
 				let ready_poll = self.ready_poll.clone();
 
 				async move {
+					// We keep track of everything we prune so that later we won't add
+					// tranactions with those hashes from the retracted blocks.
+					let mut pruned_log = HashSet::<ExtrinsicHash<PoolApi>>::new();
+
 					// If there is a tree route, we use this to prune known tx based on the enacted
 					// blocks.
 					if let Some(ref tree_route) = tree_route {
-						future::join_all(
+						for enacted_log in future::join_all(
 							tree_route
 								.enacted()
 								.iter()
-								.map(|h|
```

Comment:
Than you can revert all these changes here and just do:
```
join_all().await.into_iter().for_each(|t| pruned_log.insert(t));
```

Reply:
I know, I just wanted to reduce allocations if possible (not here, but below, for example)